### ISSUE TITLE
Add app to f-droid.org app store

### ISSUE THREAD

**Issue by guysoft** [thielepaul/photoprism-mobile#40]
The FOSS app repository is a good place to host the app, and will also draw the right kind of crowed to use it.
https://f-droid.org/

You can add a request at:
https://gitlab.com/fdroid/rfp/issues

**Comment by spaibs** [comment]
Yes, we will add our app to f-droid in the future.

**Comment by guysoft** [comment]
Why not now? What is blocking addition?

**Comment by spaibs** [comment]
Hm, does it make sense to publish an unfinished app?

**Comment by guysoft** [comment]
I guess it would make sense when you start tagging versions in the repo.
So perhaps not yet. I would suggest keeping the issue open and if you have an 1.0/beta milestone connect it to that

**Comment by spaibs** [comment]
Sounds like a good plan!

**Comment by dennorske** [comment]
I am excited for testing it. Any way I can already have a peak on it?

**Comment by thielepaul** [comment]
> I am excited for testing it. Any way I can already have a peak on it?

Sure, if you have an Android smartphones you can just download the apk that is built for every commit on the master branch: https://github.com/photoprism/photoprism-mobile/releases/download/latest/photoprism.apk

If you have an iPhone and a Mac you can build the app yourself.

**Comment by dessalines** [comment]
It'd be nice to have this on f-droid.

**Comment by SchoolGuy** [comment]
So although we cannot use F-Droid to publush Beta Releases (see: https://gitlab.com/fdroid/fdroidserver/-/issues/161), we could do this on Google Play. I know this because I have done it. Google Play also allows Beta/Alpha/etc. for a specified number of people or for open Beta. I don't love being tied to Google Play but it is a good option.

The other option would be to publish *three* apps in F-Droid:

- A stable version called "Photoprism Mobile"
- A beta version called "Photoprism Mobile Beta"
- A nightly version called "Photoprism Mobile Nightly"

With https://docs.fastlane.tools/ we could automate this so we would require next to no interaction. Also this has the advantage that other app stores integrate with Fastlane. So the iOS publishing would also be solved at the same time if I am informed correctly.

**Comment by spupuz** [comment]
please publish it on google play store i'm in for testing the app

**Comment by IzzySoft** [comment]
> So although we cannot use F-Droid to publush Beta Releases

Please stand by. In about 22h, [check here](https://apt.izzysoft.de/fdroid/index/apk/org.photoprism.photoprism). Please remember to increase `versionCode` with each release (ideally, adjust `versionName` as well and keep it in sync with the tag name – like `versionName: "1.0.0"` => tag `v1.0.0` or `beta-1.0.0` or `1.0.0-beta` – the "beta" can be used then to separate them from the stables).

**Comment by SchoolGuy** [comment]
I have successfully downloaded the App. Login into my personal Instance was successful!

**Comment by IzzySoft** [comment]
So the app wasn't updated in my repo for about a year now – `versionCode` still remained unchanged so it was assumed to be the same version… :scream: Just pulled the current version manually today.

As it's already been 2 years, will "proper tags" become available soon? Those would help keeping the app updated in my repo automatically, even if there'd be only 1 per month (and provided the `versionCode`… well).

**Comment by thielepaul** [comment]
> As it's already been 2 years, will "proper tags" become available soon? Those would help keeping the app updated in my repo automatically, even if there'd be only 1 per month (and provided the versionCode… well).
 
I do not think this app will anytime soon (maybe never) get to a state that I would feel it makes sense to "release" it. Already now I do not really have time and motivation to read through the issues here and reply to them.
If this would help you, I can create a `v0.x.0` tag from master from time to time if this does help you?
If you need some other stuff (like the version number in the actual app or so) feel free to submit a PR that automates this. If you build the app yourself, I guess you can just use the `--build-name` argument for `flutter build`.

**Comment by IzzySoft** [comment]
Yes @thielepaul – a `v0.x.0` tag from time to time would indeed be very helpful (if `versionCode` is increased and `versionName` adjusted), thanks a lot! I understand it's hard to imagine one's own baby got mature (I remember how hard it was on me to finally declare a 1.0.0 or 2.0.0 on my stuff), but never lose hope :smile: Can you please let me know when the first such tag is available, so I can see to configure my update checker to match your definitions?

Thanks for the other hints, too – but I'm neither an Android dev nor do I have the proper environment set up to build stuff myself…

**Comment by thielepaul** [comment]
no promises for now, setting this up just has no high priority for me. If I will finally do it, I'll try to remember to write a note here, so you will get a notification

**Comment by IzzySoft** [comment]
Sure, thanks – and no pressure from my end either. It's ready when it's ready. Looking forward to your next ping then!

Btw: should you be interested in a starter package for [fastlane metadata](https://gitlab.com/-/snippets/1895688) (F-Droid will want this – and it gives you the control of how your app will be presented, with description and screenshots etc), just let me know and I gladly provide you with a PR of what I've set up here to list your app. So I might have a little thing to give back as compensation :smile:

**Comment by thielepaul** [comment]
@IzzySoft 
From now on, I'll try to remember to tag the master when there are notable changes and put the matching version in the `pubspec.yaml`. I created a `v0.2.2`today. Let me know if it works for you like that and if not just submit a PR with the changes you need :)

**Comment by IzzySoft** [comment]
@thielepaul tag looks good! Re-enabled automated updates again then, and also told my updater to stick to the "release tags" (i.e. to ignore the pre-release ones) – which means you could use pre-releases for testing purposes, which my updater would not fetch, and "normal" releases for the wider audience which then would be included with my repo (and thus must have `versionCode` increased).

Are you interested in the Fastlane structures then? I could provide you what is set up with my repo currently, so you could be in control on those details (e.g. updating details, include per-release changelogs).

PS: Your Readme states something about incompatibilities. Not sure if that still applies with the update, and if it should be mentioned with the app description.

**Comment by thielepaul** [comment]
> @thielepaul tag looks good! Re-enabled automated updates again then, and also told my updater to stick to the "release tags" (i.e. to ignore the pre-release ones) – which means you could use pre-releases for testing purposes, which my updater would not fetch, and "normal" releases for the wider audience which then would be included with my repo (and thus must have `versionCode` increased).

thats how it currently works, every commit on master will create a "latest" pre-release. Whenever I manually tag it is a normal release.

> Are you interested in the Fastlane structures then? I could provide you what is set up with my repo currently, so you could be in control on those details (e.g. updating details, include per-release changelogs).

I guess, I'd leave it they way it is for now, but thanks for the offer!

> PS: Your Readme states something about incompatibilities. Not sure if that still applies with the update, and if it should be mentioned with the app description.

Yes, the incompatibilities with the upstream photoprism are still there so this app only works if using the docker images built from my photoprism fork.

**Comment by IzzySoft** [comment]
> thats how it currently works

great, then this matches :smiley:

Fastlane: call me on it when you want it then :wink:

> the incompatibilities with the upstream photoprism are still there

Then the app description better covers that, to avoid disappointments, don't you think? Please [check here](https://apt.izzysoft.de/fdroid/index/apk/org.photoprism.photoprism) and let me know what to amend (hint: that's one of the things you could accomplish here yourself with Fastlane structures :see_no_evil:). Some warning at the end?

**Comment by thielepaul** [comment]
Maybe something like this section in the readme: https://github.com/thielepaul/photoprism-mobile#currently-incompatible-with-photoprism-warning

Also the notes at the top of the readme should be there in the description: https://github.com/thielepaul/photoprism-mobile#flutter-app-for-photoprism-ios-and-android

The note about the PhotoPrism trademark probably is important, too: https://github.com/thielepaul/photoprism-mobile#photoprism-is-a-registered-trademark

**Comment by IzzySoft** [comment]
Thanks! All integrated now (becomes visible with the next sync tomorrow). Skimming the linked PR I guess it will take quite a while before it will be merged…
